conan_provider.cmake

```diff
@@ -280,12 +280,20 @@ macro(conan_provide_dependency package_name)
     endif()
 
     get_property(CONAN_GENERATORS_FOLDER GLOBAL PROPERTY CONAN_GENERATORS_FOLDER)
-    list(FIND CMAKE_PREFIX_PATH "${CONAN_GENERATORS_FOLDER}" index)
-    if(${index} EQUAL -1)
-        list(PREPEND CMAKE_PREFIX_PATH "${CONAN_GENERATORS_FOLDER}")
+
+    set(_find_args "${ARGN}")
```

Comment:
A couple of comments describing what is being done, and a link to the CMake docs explaining the 2 ``find_package()`` calls approach would be useful here.

Reply:
Added more comments!